flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/rpc/RpcCheckpointResponder.java

```diff
@@ -55,7 +56,16 @@ public void declineCheckpoint(
             long checkpointId,
             Throwable cause) {
 
+        // TODO the passed parameter 'cause' is actually always instance of CheckpointException,
+        //  we should change the interfaces to narrow all declined checkpoint's throwable to
+        // CheckpointException.
```

Comment:
IIUC, the same TODO was fixed in the corresponding PR for 1.13.
What's the reasoning behind not fixing it for 1.12?

Reply:
I don't want to break the API compatibility to refactor APIs in minor version of Flink, and we could still achieve the effect to make job could fail if async checkpoint failed without refactoring APIs.